[BUG] udf-examples module fails on `mvn compile` and `mvn test`

**Describe the bug**
 `mvn compile` and `mvn test` are failing due to the error as below

```
[INFO] --- scala-maven-plugin:4.3.0:compile (scala-compile-first) @ rapids-4-spark-udf-examples_2.12 ---
[INFO] Using incremental compilation using Mixed compile order
[INFO] Compiler bridge file: /home/liangcail/.sbt/1.0/zinc/org.scala-sbt/org.scala-sbt-compiler-bridge_2.12-1.3.1-bin_2.12.8__52.0-1.3.1_20191012T045515.jar
[INFO] Compiling 2 Scala sources and 7 Java sources to /home/liangcail/work/projects/on_github/spark-rapids/udf-examples/target/classes ...
[ERROR] [Error] /home/liangcail/work/projects/on_github/spark-rapids/udf-examples/src/main/scala/com/nvidia/spark/rapids/udf/scala/URLDecode.scala:22: object RapidsUDF is not a member of package com.nvidia.spark
[ERROR] [Error] /home/liangcail/work/projects/on_github/spark-rapids/udf-examples/src/main/scala/com/nvidia/spark/rapids/udf/scala/URLDecode.scala:30: not found: type RapidsUDF
[ERROR] [Error] /home/liangcail/work/projects/on_github/spark-rapids/udf-examples/src/main/scala/com/nvidia/spark/rapids/udf/scala/URLEncode.scala:22: object RapidsUDF is not a member of package com.nvidia.spark
[ERROR] [Error] /home/liangcail/work/projects/on_github/spark-rapids/udf-examples/src/main/scala/com/nvidia/spark/rapids/udf/scala/URLEncode.scala:30: not found: type RapidsUDF
[ERROR] four errors found
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for RAPIDS Accelerator for Apache Spark Root Project 21.12.0-SNAPSHOT:
[INFO] 
[INFO] RAPIDS Accelerator for Apache Spark Root Project ... SUCCESS [  0.636 s]
[INFO] RAPIDS Accelerator for Apache Spark SQL Plugin ..... SUCCESS [ 26.145 s]
[INFO] RAPIDS Accelerator for Apache Spark Shuffle Plugin . SUCCESS [  4.883 s]
[INFO] RAPIDS Accelerator for Apache Spark SQL Plugin Shims SUCCESS [  0.037 s]
[INFO] RAPIDS Accelerator for Apache Spark SQL Plugin Spark 3.0.1 Shim SUCCESS [  2.735 s]
[INFO] RAPIDS Accelerator for Apache Spark Scala UDF Plugin SUCCESS [  5.490 s]
[INFO] RAPIDS Accelerator for Apache Spark Aggregator ..... SUCCESS [  0.033 s]
[INFO] RAPIDS Accelerator for Apache Spark Distribution ... SUCCESS [  0.279 s]
[INFO] RAPIDS Accelerator for Apache Spark UDF Examples ... FAILURE [  1.341 s]
[INFO] RAPIDS Accelerator for Apache Spark Tests .......... SKIPPED
[INFO] rapids-4-spark-integration-tests_2.12 .............. SKIPPED
[INFO] rapids-4-spark-api-validation ...................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

```

**Steps/Code to reproduce bug**
Fetch the latest commit and run `mvn test` under the project root path.

**Expected behavior**
`mvn test` and `mvn compile` should run successfully.

**Additional context**
`mvn test -DwildcardSuites=XXX` is quite useful during local development. Hope we can support it.

git log output:
```
commit bfaf73b2b96a156c4c321c2f0cbe36c6615574ad (NVIDIA/branch-21.12, branch-21.12)
Merge: 81186e9b 5733da23
Author: Jason Lowe <jlowe@nvidia.com>
Date:   Thu Oct 28 16:09:26 2021 -0500

    Merge pull request #3950 from gerashegalov/fix-merge
    
    Fix failed auto-merge #3939

commit 5733da23ca9c471daa42b2ff2e5421072e2ac0b5
Author: Gera Shegalov <gera@apache.org>
Date:   Thu Oct 28 11:13:31 2021 -0700

    revert doc update

commit 81186e9be1f30f1f8be7eddbcb3d163f54ee238f
Author: Robert (Bobby) Evans <bobby@apache.org>
Date:   Thu Oct 28 12:56:42 2021 -0500

    Document compatability of operations with side effects. (#3946)
    
    Signed-off-by: Robert (Bobby) Evans <bobby@apache.org>

```

--- a/udf-examples/pom.xml
+++ b/udf-examples/pom.xml
@@ -58,8 +58,14 @@
     </dependency>
     <dependency>
         <groupId>com.nvidia</groupId>
-        <artifactId>rapids-4-spark_${scala.binary.version}</artifactId>
+        <!--
+        This should depend on rapids-4-spark_${scala.binary.version} instead, but that dependency
+        only exists after the package phase. External projects should depend on
+        rapids-4-spark_${scala.binary.version}.
+        -->
+        <artifactId>rapids-4-spark-sql_${scala.binary.version}</artifactId>
         <version>${project.version}</version>
+        <classifier>${spark.version.classifier}</classifier>
         <scope>provided</scope>
     </dependency>
   </dependencies>

